tests/dragonfly/replication_test.py

```diff
@@ -2411,3 +2411,66 @@ async def test_replicate_old_master(
     assert await c_replica.execute_command("get", "k1") == "v1"
 
     await disconnect_clients(c_master, c_replica)
+
+
+@pytest.mark.asyncio
+async def test_empty_hash_map_replicate_old_master(df_factory):
+    cpu = platform.processor()
+    if cpu != "x86_64":
+        pytest.skip(f"Supported only on x64, running on {cpu}")
+
+    dfly_version = "v1.21.2"
+    released_dfly_path = download_dragonfly_release(dfly_version)
+
```

Comment:
maybe leave a comment somwhere that v1.21.2 (and before) has a known issue with hsetex not deleting the key, etc... because the person who'll be reading it in a year  has to follow a bunch of git blame links to even find out what this is about

Reply:
nit: the usual `@dfly_args({"proactor_threads": 1})` reminder (we really need to set a global limit)